File: Algorithms/02.py
def searchVertical(i, j, wStar):
    hStar = 0
    for k in range(i, -1, -1):
        if myTable[k][j] == '*':
            hStar += 1
    if hStar >= 2 * wStar:
        return True


n, m = map(int, input().split())

myTable = []
for i in range(n):
    myTable.append(list(input()))

result = 0
for i in range(n - 1, -1, -1):
    wStar = 0
    for j in range(m - 1, -1, -1):
        if myTable[i][j] == '*':
            wStar += 1
            if wStar >= 2:
                for t in range(2, wStar + 1):
                    if searchVertical(i, j, t) is True:
                        result += 1

print(result)

File: Algorithms/02.js
let nm = readline().split(' ');
let n = parseInt(nm[0]);
let m = parseInt(nm[1]);

let myTable = new Array();
for (let i = 0; i < n; i++) {
    myTable.push(readline().split(''));
}

var result = 0;
for (let i = n - 1; i >= 0; i--) {
    let wStar = 0;
    for (let j = m - 1; j >= 0; j--) {
        if (myTable[i][j] == '*') {
            wStar++;
            if (wStar >= 2) {
                for (let t = 2; t <= wStar; t++)
                {
                    searchVertical(i, j, t);
                }
            }
        }
    }
}

function searchVertical(i, j, wStar) {
    let hStar = 0;
    for (let k = i; k >= 0; k--) {
        if (myTable[k][j] == '*') {
            hStar++;
        }
    }
    if (hStar >= 2 * wStar) {
        result++;
    }
}

print(result)